1Password/shell-plugins
# Add Terraform plugin

## Overview

This PR introduces the definition logic for enabling a Terraform shell plugin and similar plugins to be supported. 

It also introduces the Terraform shell plugin, as an example

## Thought process
Credential Usage SDK definition is to be modified to support a SelectFrom CredentialSelection field. This struct defines how a credential selection is to take place in op. This is mutually exclusive with the Name Plugin and Provisioner fields of CredentialUsage.

Additionally the Credential Usage struct will be specified with an Optional field to determine at init step whether the executable can be run without this credential or not.
NeedsAuth will also be added at the level of a credential usage to enable more granular selection of when to provision a credential or not based on the use case.
## Type of change
 


## Related Issue(s)
  

* Resolves: #140 

## How To Test
This can only be tested by internal contributors.

1. `op plugin init terraform` - configure with whatever credentials you want
2. `terraform init` in a new dir
3.  make sure the `main.tf` uses the providers for which you've configured the credentials
4. `terraform plan` - this should work without saying that no credentials were found, even though no credentials are on disk.




## Changelog
Authenticate the Terraform CLI using Touch ID and other unlock options with 1Password Shell Plugins.
